Sources/CoreDataRepository/CoreDataRepository+Create.swift

```diff
@@ -0,0 +1,39 @@
+// CoreDataRepository+Create.swift
+// CoreDataRepository
+//
+//
+// MIT License
+//
+// Copyright © 2024 Andrew Roan
+
+import CoreData
+
+extension CoreDataRepository {
+    /// Create an instance in the store.
+    @inlinable
+    public func create<Model>(
+        _ item: Model,
+        transactionAuthor: String? = nil
+    ) async -> Result<Model, CoreDataError> where Model: WritableUnmanagedModel, Model: FetchableUnmanagedModel {
```

Comment:
Curious, any reason to use the syntax `Result<Model, CoreDataError> where Model: WritableUnmanagedModel, Model: FetchableUnmanagedModel` over `Result<Model, CoreDataError> where Model: WritableUnmanagedModel & FetchableUnmanagedModel`? I thought the second method was more concise and commonly used in Swift.

Reply:
Personally, I prefer the single conformance per clause format. I think this format is more consistently formattable and readable because each clause is the same shape instead of some having two conformances separated by `&`.